security/tutorials/calico-policy.md

```diff
@@ -0,0 +1,345 @@
+---
+title: Calico policy tutorial
+description: Learn how to create more advanced Calico network policies (namespace, allow and deny all ingress and egress).
+canonical_url: "/security/tutorials/calico-policy-advanced"
+---
+
+Calico Network Policies _extend_ the functionalities of Kubernetes Network Policies. To demonstrate this, this page will walk through the [Kubernetes Advanced NetworkPolicy](kubernetes-policy-advanced) tutorial but with everything implemented using Calico policies. This serves as a good example in highlighting the syntactical differences between the two type of policies while demonstrating Calico Network Policy's flexibility.
+
+### Requirements
+
+- Calico v2.6.1+ with Kubernetes 1.8+
+- A working Kubernetes cluster and access to it using kubectl and calicoctl
+- Your Kubernetes nodes have connectivity to the public internet
+- You are familiar with [Kubernetes NetworkPolicy](kubernetes-policy-basic) or [Kubernetes NetworkPolicy Advanced](kubernetes-policy-advanced)
+
+### Tutorial flow
+
+1. Create the Namespace and Nginx Service
+1. Deny all ingress traffic
+1. Allow ingress traffic to Nginx
+1. Deny all egress traffic
+1. Allow egress traffic to kube-dns
+1. Cleanup Namespace
+
+### 1. Create the namespace and nginx service
+
+We'll use a new namespace for this guide. Run the following commands to create it and a plain nginx service listening on port 80.
+
+```bash
+kubectl create ns advanced-policy-demo
+kubectl run --namespace=advanced-policy-demo nginx --replicas=2 --image=nginx
+kubectl expose --namespace=advanced-policy-demo deployment nginx --port=80
+```
+
+#### Verify access - allowed all ingress and egress
+
+Open up a second shell session which has `kubectl` connectivity to the Kubernetes cluster and create a busybox pod to test policy access. This pod will be used throughout this tutorial to test policy access.
+
+```bash
+kubectl run --namespace=advanced-policy-demo access --rm -ti --image busybox /bin/sh
+```
+
+This should open up a shell session inside the `access` pod, as shown below.
```

Comment:
Use imperative, unless you really mean "should" -- it may or may not do xyz...

This opens up a shell...

Reply:
I changed alot of the sentences to use imperative! :D